webapp/src/ecs/ECS.test.ts

```diff
@@ -0,0 +1,45 @@
+import ECS from './ECS';
+
+test('Test String Identifier', () => {
+    const ecs = new ECS();
+    // Create the component
+    const defaultState = {
+        hp: 100,
+        stamina: 100,
+    }
+    ecs.addComponent('Player', {
+        name: 'Player',
+        description: 'Players health and stamina',
+        defaultState
+    });
+
+    // Create entity
+    // Use 'Player' as id
+    const playerOne = ecs.createEntity(['Player']);
+
+    // Check that the saved component data is the same data, but not the same object
+    expect(ecs.getComponentData('Player', playerOne)).not.toBe(defaultState);
+    expect(ecs.getComponentData('Player', playerOne)).toEqual(defaultState);
+});
+
+test('Test Number Identifier', () => {
```

Comment:
You set this component again, which is kind of unnecessary and WET. For future tests, check out the setup function [https://jestjs.io/docs/setup-teardown](https://jestjs.io/docs/setup-teardown)

Reply:
Good tip, I did not know that was a thing.